pkg/controller/atlas/client_test.go

```diff
@@ -17,3 +19,24 @@ func TestClientUserAgent(t *testing.T) {
 	r.NoError(err)
 	r.Contains(c.UserAgent, version.Version)
 }
+
+type TestClientBuilder struct {
+	used bool
+}
+
+func (tcb *TestClientBuilder) NewClientFn() *http.Client {
+	tcb.used = true
+	return &http.Client{Transport: http.DefaultTransport}
+}
+
+func TestCustomClient(t *testing.T) {
+	tcb := &TestClientBuilder{used: false}
+	atlas.CustomHTTPClientFn = tcb.NewClientFn
+	defer func() {
```

Comment:
The presence of this in your test shows how risky is the current approach.

Reply:
fixed